src/ng/q.js

```diff
@@ -636,6 +636,34 @@ function qFactory(nextTick, exceptionHandler, errorOnUnhandledRejections) {
     return deferred.promise;
   }
 
+  /**
+   * @ngdoc method
+   * @name $q#race
+   * @kind function
+   *
+   * @description
+   * Returns a promise that is resolved similarly as the first promise to resolve.
+   *
+   * @param {Array.<Promise>|Object.<Promise>} promises An array or hash of promises.
+   * @returns {Promise} Returns a promise that will be resolved as soon as the first promise in
+   *   `promises` is resolved. The resolved value will be the same as the value of the first
+   *   resolved promise.
+   */
+
+  function race(promises) {
+    var deferred = new Deferred();
+
+    forEach(promises, function(promise) {
+      when(promise).then(function(value) {
```

Comment:
> Indeed the code inside the `forEach` callback (`when(...).then(...)`) won't throw.

I think what @petebacondarwin meant is more on whether `forEach` will throw on its own (i.e., due to bad input), no? But as mentioned, looking at how `forEach` is implemented, I don't think that will happen. Furthermore, since `when(...).then(...)` is also safe, I guess this should be safe. :smile:

> BTW, this can be abbreviated as:
>
> `when(promise).then(deferred.resolve, deferred.reject);`

Good point. Thanks for pointing this out!